configdb/hardware/api_views.py

```diff
@@ -9,11 +10,40 @@
 )
 
 
+class CustomViewSchema(AutoSchema):
+    """
+    Class to generate OpenAPI schema from views
+
+    Users of this class may specify custom filter annotations to override those
+    generated by Django Rest Framework.
+    """
+    def __init__(self, custom_filter_annotations=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.custom_filter_annotations = custom_filter_annotations
+
+    def get_filter_parameters(self, path, method):
+        parameters = super().get_filter_parameters(path, method)
+        # If a set of additional filter annotations were provided, override what Django has generated
+        if self.custom_filter_annotations is not None:
+            for annotation in self.custom_filter_annotations:
+                for parameter in parameters:
+                    if parameter['name'] == annotation['name']:
+                        parameter['description'] = annotation['description']
+                        # if a type has been specified, add that to the OpenAPI schema. String is the default
+                        if annotation.get('type'):
+                            parameter['schema'] = {'type': annotation['type']}
+        return parameters
+
+
 class FilterableViewSet(viewsets.ModelViewSet):
     filter_backends = (DjangoFilterBackend,)
 
 
 class SiteViewSet(FilterableViewSet):
+    custom_filter_annotations = [{'name': 'name', 'description': 'Site name'},
```

Comment:
A list of dictionaries can be provided to add more detail to Django's autogenerated query param lists.

Currently we support query description and query type (integer, string, etc...)